truffle/contracts/CommonPot.sol

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity >=0.7.0 <0.9.0;
+
+import "./Allowance.sol";
+import "@openzeppelin/contracts/utils/math/SafeMath.sol";
+import "@openzeppelin/contracts/utils/Strings.sol";
+
+contract CommonPot is Allowance {
+
+    using SafeMath for uint;
+    using Strings for uint256;
+
+    address[] private contributors ;
+    mapping(address => uint) private balances;
+    uint private totalBalance = 0;
+
+    event Deposit(address sender, uint amount);
+    event Withdrawal(address receiver, string amount);
+
+    receive() external payable {
+        require(msg.value > 0, "No money sent");
+        balances[msg.sender] = SafeMath.add(balances[msg.sender], msg.value);
+        totalBalance = SafeMath.add(totalBalance, msg.value);
+        if (!contributorsPresent(msg.sender)) {
+            contributors.push(msg.sender);
+        }
+        emit Deposit(msg.sender, msg.value);
+    }
+
+    function withdraw() public isOwner {
+        //prevent someone who has no interest from emptying contract's balance
+        require(balances[msg.sender] > 0, "Insufficient funds");
+        uint256 quotient;
+        uint256 remainder;
+        string memory result;
+        for (uint i = 0; i < contributors.length; i++) {
+            (quotient, remainder, result) = division(totalBalance, balances[contributors[i]], 10);
+            emit Withdrawal(contributors[i], result);
+            balances[contributors[i]] = 0;
+        }
+        totalBalance = 0;
+    }
+
+    function payWithPot(address to, uint amount) payable public {
+        require(totalBalance > amount, "Insufficient funds");
+        require(msg.sender == isOwner(), "Only owner can pay with the pot");
+        payable(to).transfer(amount);
```

Comment:
might need to update `totalBalance` since functions like `getCurrentGlobalBalance()` and `withdraw()` relies on it